src/main/java/codechicken/nei/util/NBTJson.java

```diff
@@ -20,26 +17,21 @@
 import net.minecraft.nbt.NBTTagShort;
 import net.minecraft.nbt.NBTTagString;
 
-import java.io.ByteArrayOutputStream;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import java.util.stream.IntStream;
 
 public class NBTJson {
-    private static final Pattern numberPattern = Pattern.compile("^([-+]?[\\d]+\\.?[0-9]*)([bBsSlLfFdD]?)$");
-    private static final JsonParser parser = new JsonParser();
-    private static final Gson gson = new GsonBuilder().create();
+    private static final Pattern numberPattern = Pattern.compile("^([-+]?\\d+\\.?\\d*E*\\d*)([bBsSlLfFdD]?)$");
 
     public static String toJson(NBTTagCompound tag) {
         return toJson(toJsonObject(tag));
     }
 
     public static String toJson(JsonElement json) {
-        final String jsonString = json.toString();
-        return gson.toJson(parser.parse(jsonString));
+        return json.toString();
```

Comment:
I did not find the reason for turning an object into a string after in an object and then back into a string

Reply:
Probably testing roundtrip compatability at some point and forgot to remove it